include/powsybl/iidm/LoadingLimitsAdder.hpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef POWSYBL_IIDM_LOADINGLIMITSADDER_HPP
+#define POWSYBL_IIDM_LOADINGLIMITSADDER_HPP
+
+#include <powsybl/iidm/LoadingLimits.hpp>
+#include <powsybl/iidm/OperationalLimitsAdder.hpp>
+#include <powsybl/stdcxx/math.hpp>
+#include <powsybl/stdcxx/optional.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+class OperationalLimitsHolder;
+
+template <typename L, typename A>
+class LoadingLimitsAdder : public OperationalLimitsAdder<L> {
+public:
+    class TemporaryLimitAdder {
+    public:
+        TemporaryLimitAdder(OperationalLimitsHolder& owner, LoadingLimitsAdder<L, A>& parent);
+
+        LoadingLimitsAdder<L, A>& endTemporaryLimit();
+
+        TemporaryLimitAdder& ensureNameUnicity();
+
+        TemporaryLimitAdder& setAcceptableDuration(unsigned long duration);
+
+        TemporaryLimitAdder& setFictitious(bool fictitious);
+
+        TemporaryLimitAdder& setName(const std::string& name);
+
+        TemporaryLimitAdder& setValue(double value);
+
+    private:
+        void checkAndGetUniqueName();
```

Comment:
There is something strange with this design but you do exactly what is done in Java. From my opinion, a method that `check and get` should take a parameter and return something...